presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateFloatPercentileAggregations.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.DigestAndPercentileState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.type.SqlType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.stats.QuantileDigest;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.FloatType.FLOAT;
+import static com.facebook.presto.testing.AggregationTestUtils.generateInternalAggregationFunction;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static com.google.common.base.Preconditions.checkState;
+
+@AggregationFunction("approx_percentile")
+public class ApproximateFloatPercentileAggregations
+{
+    public static final InternalAggregationFunction FLOAT_APPROXIMATE_PERCENTILE_AGGREGATION =
+            generateInternalAggregationFunction(
+                    ApproximateFloatPercentileAggregations.class,
+                    FLOAT.getTypeSignature(),
+                    ImmutableList.of(FLOAT.getTypeSignature(), DOUBLE.getTypeSignature()));
+    public static final InternalAggregationFunction FLOAT_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION =
+            generateInternalAggregationFunction(ApproximateFloatPercentileAggregations.class,
+                    FLOAT.getTypeSignature(),
+                    ImmutableList.of(FLOAT.getTypeSignature(), BIGINT.getTypeSignature(), DOUBLE.getTypeSignature()));
+    public static final InternalAggregationFunction FLOAT_APPROXIMATE_PERCENTILE_WEIGHTED_AGGREGATION_WITH_ACCURACY =
+            generateInternalAggregationFunction(ApproximateFloatPercentileAggregations.class,
+                    FLOAT.getTypeSignature(),
+                    ImmutableList.of(FLOAT.getTypeSignature(), BIGINT.getTypeSignature(), DOUBLE.getTypeSignature(), DOUBLE.getTypeSignature()));
+
+    private ApproximateFloatPercentileAggregations() {}
+
+    @InputFunction
+    public static void input(DigestAndPercentileState state, @SqlType(StandardTypes.FLOAT) long value, @SqlType(StandardTypes.DOUBLE) double percentile)
+    {
+        ApproximateLongPercentileAggregations.input(state, value, percentile);
```

Comment:
It looks like this relies on our internal representation of `FLOAT` sorting in the same order as `long`. Does IEEE 754 work that way?